docs: Update configuration file pages (#16509)
eslint/eslint: docs/src/user-guide/configuring/configuration-files-new.md

> This is an experimental feature. To opt-in, place a `eslint.config.js` file in the root of your project or set the `ESLINT_USE_FLAT_CONFIG` environment variable to `true`. To opt-out, even in the presence of a `eslint.config.js` file, set the environment variable to `false`. If you are using the API, you can use the configuration system described on this page by using the `FlatESLint` class, the `FlatRuleTester` class, or by setting `configType: "flat"` in the `Linter` class.

Reviewer: ```suggestion
This is an experimental feature. To opt-in, place an `eslint.config.js` file in the root of your project or set the `ESLINT_USE_FLAT_CONFIG` environment variable to `true`. To opt-out, even in the presence of an `eslint.config.js` file, set the environment variable to `false`. If you are using the API, you can use the configuration system described on this page by using the `FlatESLint` class, the `FlatRuleTester` class, or by setting `configType: "flat"` in the `Linter` class.
```

Reviewer 2: Here "To opt-in, place an `eslint.config.js` file in the root" doesn’t sound good right? It can be "To opt-in, place `eslint.config.js` file " or "To opt-in, place a `eslint.config.js` file"

Reviewer: Usually `an` goes before the next word that starts with vowels .Thats why I replaced it with `an`.

#### Reporting unused disable directives

> Disable directives such as `/*eslint-disable*/` and `/*eslint-disable-next-line*/` are used to disable ESLint rules around certain portions of code. As code changes, it's possible for these directives to no longer be needed because the code has changed in such a way that the rule is longer be triggered. You can enable reporting of these unused disable directives by setting the `reportUnusedDisableDirectives` option to `true`, as in this example:

Author: ```suggestion
Disable directives such as `/*eslint-disable*/` and `/*eslint-disable-next-line*/` are used to disable ESLint rules around certain portions of code. As code changes, it's possible for these directives to no longer be needed because the code has changed in such a way that the rule is no longer triggered. You can enable reporting of these unused disable directives by setting the `reportUnusedDisableDirectives` option to `true`, as in this example:
```

> Disable directives such as `/*eslint-disable*/` and `/*eslint-disable-next-line*/` are used to disable ESLint rules around certain portions of code. As code changes, it's possible for these directives to no longer be needed because the code has changed in such a way that the rule is no longer triggered. You can enable reporting of these unused disable directives by setting the `reportUnusedDisableDirectives` option to `true`, as in this example:

Reviewer: ```suggestion
Disable directives such as `/*eslint-disable*/` and `/*eslint-disable-next-line*/` are used to disable ESLint rules around certain portions of code. As code changes, it's possible for these directives to no longer be needed because the code has changed in such a way that the rule is no longer triggered. You can enable reporting of these unused, disabled directives by setting the `reportUnusedDisableDirectives` option to `true`, as in this example:
```

Author: ```suggestion
Disable directives such as `/*eslint-disable*/` and `/*eslint-disable-next-line*/` are used to disable ESLint rules around certain portions of code. As code changes, it's possible for these directives to no longer be needed because the code has changed in such a way that the rule is no longer triggered. You can enable reporting of these unused disabled directives by setting the `reportUnusedDisableDirectives` option to `true`, as in this example:
```

> Disable directives such as `/*eslint-disable*/` and `/*eslint-disable-next-line*/` are used to disable ESLint rules around certain portions of code. As code changes, it's possible for these directives to no longer be needed because the code has changed in such a way that the rule is no longer triggered. You can enable reporting of these unused disabled directives by setting the `reportUnusedDisableDirectives` option to `true`, as in this example:

Reviewer 3: ```suggestion
Disable directives such as `/*eslint-disable*/` and `/*eslint-disable-next-line*/` are used to disable ESLint rules around certain portions of code. As code changes, it's possible for these directives to no longer be needed because the code has changed in such a way that the rule is no longer triggered. You can enable reporting of these unused disable directives by setting the `reportUnusedDisableDirectives` option to `true`, as in this example:
```

Directives are not disabled, they're "disable directives".

];
```

> This configuration ensures that the Babel parser, rather than the default Espree parser, is used to parse all files ending with `.js` and `.mjs`.

Reviewer: ```suggestion
This configuration ensures that the Babel parser, rather than the default Espree parser is used to parse all files ending with `.js` and `.mjs`.
```

Author: it is correct to keep this comma here. "rather than the default Espree parser" is a [non-essential aside](https://owl.purdue.edu/owl/general_writing/punctuation/commas/commas_with_nonessential_elements.html), which requires using a comma.

### Configuring shared settings

> ESLint supports adding shared settings into configuration files. Plugins use `settings` to specify information that should be shared across all of its rules. When you add a `settings` object to a configuration object, it is supplied to every rule executed by that configuration object. This may be useful if you are adding custom rules and want them to have access to the same information. Here's an example:

Author: ```suggestion
ESLint supports adding shared settings into configuration files. When you add a `settings` object to a configuration object, it is supplied to every rule executed by that configuration object. Plugins can use `settings` to specify the information that should be shared across all of their rules. This may be useful if you are adding custom rules and want them to have access to the same information.  By convention, plugins namespace the settings they are interested in to avoid collisions with others. Here's an example:
```

> ESLint supports adding shared settings into configuration files. Plugins use `settings` to specify information that should be shared across all of its rules. When you add a `settings` object to a configuration object, and it is supplied to every rule executed by that configuration object. This may be useful if you are adding custom rules and want them to have access to the same information. Here's an example:

Reviewer 4: ```suggestion
ESLint supports adding shared settings into configuration files. Plugins use `settings` to specify the information that should be shared across all of its rules. When you add a `settings` object to a configuration object, and it is supplied to every rule executed by that configuration object. This may be useful if you are adding custom rules and want them to have access to the same information. Here's an example:
```

Author: this change looks good.

but upon re-reading the sentence, i think there's something wrong with the 'across all **its** rules' phrase. what does 'its' refer to?

@snitin315 or @nzakas, do you know?

Reviewer 4: I believe `it` is referring to the `Plugin` here.

Reviewer 5: That’s correct. Note: settings are not limited to one plugin. Every plugin receives every setting that was set in the config. It’s a convention that plugins namespace the settings they are interested in to avoid collisions with others.

Author: thanks for the clarifications. i believe i've addressed this confusion in the latest commit, [53dd21d](https://github.com/eslint/eslint/pull/16509/commits/53dd21d21609449d6fa5a965fd967fa48606b1ef).

also added the word 'the' before information that started this conversation.

> ESLint supports adding shared settings into configuration files. Plugins use `settings` to specify information that should be shared across all of its rules. When you add a `settings` object to a configuration object, and it is supplied to every rule executed by that configuration object. This may be useful if you are adding custom rules and want them to have access to the same information. Here's an example:

Reviewer: ```suggestion
ESLint supports adding shared settings into configuration files. Plugins use `settings` to specify information that should be shared across all of its rules. When you add a `settings` object to a configuration object, it is supplied to every rule executed by that configuration object. This may be useful if you are adding custom rules and want them to have access to the same information. Here's an example:
```

> ESLint supports adding shared settings into configuration files. When you add a `settings` object to a configuration object, it is supplied to every rule executed by that configuration object. By convention, plugins namespace the settings they are interested in to avoid collisions with others. Plugins can use `settings` to specify the information that should be shared across all of their rules. This may be useful if you are adding custom rules and want them to have access to the same information. Here's an example:

Reviewer 3: "it is supplied to every rule executed by that configuration object" might be misleading. It sounds like the `settings` object is only supplied to `rules: {}` from the same configuration object.

Author: @mdjermanovic 
what'd be correct, it's applied to all rules in the project?

Reviewer 5: I think you can just drop “executed by that configuration object”.

Reviewer 3: It's supplied to every rule when linting a file to which the configuration object applies, but those rules don't have to be enabled by that configuration object.

For example:

```js
module.exports = [
    {
        rules: {
            "my-plugin/my-rule": "error"
        }
    },
    {
        files: ["a.js"],
        settings: {
            data: "something"
        }
    }
];
```

When linting `a.js`, `my-plugin/my-rule` will be supplied with `{ data: "something" }` (as `context.settings`), although `my-plugin/my-rule` rule isn't enabled in the same configuration object where `settings` are defined.

Since this sentence only briefly describes what `settings` means, it could just say "it is supplied to every rule", as @nzakas suggested.

Author: thanks for the further clarification, implementing nicholas's suggestion!
